padl/transforms.py

```diff
@@ -662,33 +667,64 @@ def _pd_unpack_argument(self, arg) -> bool:
         except ValueError:
             return False
         for param in parameters:
-            if param.kind in (
+            param_kind = param.kind
+            if param_kind in (
                     param.POSITIONAL_OR_KEYWORD,
                     param.POSITIONAL_ONLY):
                 signature_count += 1
-            if param.kind == param.VAR_POSITIONAL:
+            if param_kind == param.VAR_POSITIONAL:
                 return True
         if signature_count > 1:
             return True
         return False
 
-    def pd_call_transform(self, arg, mode: Optional[Mode] = None):
+    def _pd_unpack_args_and_call(self, arg):
+        if self._pd_unpack_argument(arg):
+            return self(*arg)
+        return self(arg)
+
+    def pd_call_in_mode(self, arg, mode: Mode, ignore_grad=False):
         """Call the transform, with possibility to pass multiple arguments.
 
         :param arg: argument to call the transform with
         :param mode: The mode ("infer", "eval", "train") to perform the call with.
+        :param ignore_grad: If *True* gradient settings are ignored
         :return: Whatever the transform returns.
         """
+        no_grad = mode in ('eval', 'infer') and not ignore_grad
 
-        if mode in ('eval', 'infer'):
-            torch_context = torch.no_grad()
-        else:
-            torch_context = contextlib.suppress()
+        layers = self.pd_layers
+        if not layers:
+            no_grad = False
```

Comment:
@wuhu I added this check for the situation where `pd_forward` is not Identity but does not contain any torch layers.

Reply:
:+1: